app/services/update_interviews_provider.rb

```diff
@@ -18,16 +18,27 @@ def save!
     auth.assert_can_set_up_interviews!(application_choice: application_choice,
                                        course_option: application_choice.current_course_option)
 
-    interviews.upcoming.each do |interview|
-      next if interview.cancelled?
-      next if provider_already_associated_with_interview?(interview)
+    return if interview_list.empty?
 
-      update_provider!(interview)
+    interview_list.each { |interview| update_provider!(interview) }
+  end
+
+  def notify!
```

Comment:
```suggestion
  def notify
```
can we reserve the bang for record updates rather than sending mail